pathspider/plugins/ecn.py

```diff
@@ -6,19 +6,23 @@
 from pathspider.base import CONN_OK
 from pathspider.sync import SynchronizedSpider
 from pathspider.chains.basic import BasicChain
+from pathspider.traceroute_base import traceroute
 from pathspider.chains.tcp import TCPChain
 from pathspider.chains.tcp import TCP_SAE
 from pathspider.chains.tcp import TCP_SAEC
 from pathspider.chains.ecn import ECNChain
+from pathspider.chains.traceroute import tracerouteChain
+from pathspider.chains.trace_ecn import ECNChain_trace
 
-
-class ECN(SynchronizedSpider, PluggableSpider):
+class ECN(SynchronizedSpider, PluggableSpider, traceroute):
 
     name = "ecn"
     description = "Explicit Congestion Notification"
     version = pathspider.base.__version__
-    chains = [BasicChain, TCPChain, ECNChain]
+    chains = [BasicChain, TCPChain, ECNChain, tracerouteChain]
+    trace_chains = [ECNChain_trace]
     connect_supported = ["http", "https", "tcp", "dnstcp"]
+    traceroute_conditions = ["ecn.connectivity.broken"]
```

Comment:
We can use the internal condition "pathspider.investigate" to trigger a traceroute, instead of specifying conditions individually for plugins.